include/devicetree/pwms.h

```diff
@@ -101,6 +101,79 @@ extern "C" {
  */
 #define DT_PWMS_LABEL(node_id) DT_PWMS_LABEL_BY_IDX(node_id, 0)
 
+/**
+ * @brief Get a pwm-controller node identifier from a pwms property at an index
+ *
+ * Example devicetree fragment:
+ *
+ *     pwm1: pwm-controller@... {
+ *             label = "PWM_1";
+ *     };
+ *
+ *     pwm2: pwm-controller@... {
+ *             label = "PWM_2";
+ *     };
+ *
+ *     n: node {
+ *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
+ *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_PWMS_CTLR_BY_IDX(DT_NODELABEL(n), 0) // "PWM_1"
+ *     DT_PWMS_CTLR_BY_IDX(DT_NODELABEL(n), 1) // "PWM_2"
```

Comment:
```suggestion
 * @brief Get the node identifier for the PWM controller from a
 *        pwms property at an index
 *
 * Example devicetree fragment:
 *
 *     pwm1: pwm-controller@... { ... };
 *
 *     pwm2: pwm-controller@... { ... };
 *
 *     n: node {
 *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
 *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
 *     };
 *
 * Example usage:
 *
 *     DT_PWMS_CTLR_BY_IDX(DT_NODELABEL(n), 0) // DT_NODELABEL(pwm1)
 *     DT_PWMS_CTLR_BY_IDX(DT_NODELABEL(n), 1) // DT_NODELABEL(pwm2)
```

- brief comment change to try to match DT_GPIO_CTLR_BY_IDX more closely
- label removal inside the pwm controllers because they don't matter here, also matching DT_GPIO_CTLR_BY_IDX
- example usage change because this macro does not expand to a string, it expands to a node identifier